src/Literate.jl

```diff
@@ -262,6 +222,45 @@ function create_configuration(inputfile; user_config, user_kwargs)
     cfg["codefence"] = get(user_config, "documenter", true) ?
         ("```@example $(get(user_config, "name", cfg["name"]))" => "```") : ("```julia" => "```")
     cfg["execute"] = true
+    # Guess the package (or repository) root url
+    if (repo_slug = get(ENV, "TRAVIS_REPO_SLUG", nothing)) !== nothing
+        # GitHub repository running on Travis CI
+        edit_commit = "master" # TODO: Make this configurable like Documenter?
+        deploy_branch = "gh-pages" # TODO: Make this configurable like Documenter?
+        deploy_folder = if get(ENV, "TRAVIS_PULL_REQUEST", nothing) == "false"
+            get(ENV, "TRAVIS_TAG", get(user_config, "devurl", "dev"))
+        else
+            "previews/PR$(get(ENV, "TRAVIS_PULL_REQUEST", "##"))"
+        end
+        cfg["repo_root_url"] = "https://github.com/$(repo_slug)/blob/$(edit_commit)"
+        cfg["nbviewer_root_url"] = "https://nbviewer.jupyter.org/github/$(repo_slug)/blob/$(deploy_branch)/$(deploy_folder)"
+        cfg["binder_root_url"] = "https://mybinder.org/v2/gh/$(repo_slug)/$(deploy_branch)?filepath=$(deploy_folder)"
+    elseif (repo_slug = get(ENV, "GITHUB_REPOSITORY", nothing)) !== nothing
+        # GitHub repository running on GitHub Actions
+        edit_commit = "master" # TODO: Make this configurable like Documenter?
+        deploy_branch = "gh-pages" # TODO: Make this configurable like Documenter?
+        deploy_folder = if get(ENV, "GITHUB_EVENT_NAME", nothing) == "push"
+            if (m = match(r"^refs\/tags\/(.*)$", get(ENV, "GITHUB_REF", ""))) !== nothing
+                String(m.captures[1])
+            else
+                get(user_config, "devurl", "dev")
+            end
+        elseif (m = match(r"refs\/pull\/(\d+)\/merge", get(ENV, "GITHUB_REF", ""))) !== nothing
+            "previews/PR$(m.captures[1])"
+        else
+            "dev"
+        end
+        cfg["repo_root_url"] = "https://github.com/$(repo_slug)/blob/$(edit_commit)"
+        cfg["nbviewer_root_url"] = "https://nbviewer.jupyter.org/github/$(repo_slug)/blob/$(deploy_branch)/$(deploy_folder)"
+        cfg["binder_root_url"] = "https://mybinder.org/v2/gh/$(repo_slug)/$(deploy_branch)?filepath=$(deploy_folder)"
+    elseif (repo_url = get(ENV, "CI_PROJECT_URL", nothing)) !== nothing
+        # GitLab CI: halp plz
```

Comment:
Can probably just make this work automatically, but https://docs.gitlab.com/ee/ci/variables/predefined_variables.html don't contain any example values, so not sure which variables makes sense to use.